modules/setting/mailer.go

```diff
@@ -91,6 +90,108 @@ func newMailService() {
 		}
 	}
 
+	if MailService.MailerType == "" {
+		MailService.MailerType = "smtp"
+	}
+
+	// FIXME: DEPRECATED to be removed in v1.18.0
+	deprecatedSetting("mailer", "HOST", "mailer", "SMTP_ADDR")
+	if sec.HasKey("HOST") {
+		givenHost := sec.Key("HOST").String()
+		addr, port, err := net.SplitHostPort(givenHost)
+		if err != nil {
+			log.Fatal("Invalid mailer.HOST (%s): %v", givenHost, err)
+		}
+		MailService.SMTPAddr = addr
+		MailService.SMTPPort = port
+	}
+
+	// FIXME: DEPRECATED to be removed in v1.18.0
+	deprecatedSetting("mailer", "IS_TLS_ENABLED", "mailer", "PROTOCOL")
+	if sec.HasKey("IS_TLS_ENABLED") {
+		if sec.Key("IS_TLS_ENABLED").MustBool() {
+			MailService.Protocol = "smtps"
+		} else {
+			MailService.Protocol = "smtp+startls"
+		}
+	}
+
+	if MailService.SMTPPort == "" {
+		switch MailService.Protocol {
+		case "smtp":
+			MailService.SMTPPort = "25"
+		case "smtps":
+			MailService.SMTPPort = "465"
+		case "smtp+startls":
+			MailService.SMTPPort = "587"
+		}
+	}
+
+	if MailService.MailerType == "smtp" && MailService.Protocol == "" {
+		if strings.ContainsAny(MailService.SMTPAddr, "/\\") {
+			MailService.Protocol = "smtp+unix"
+		} else {
+			switch MailService.SMTPPort {
+			case "25":
+				MailService.Protocol = "smtp"
+			case "465":
+				MailService.Protocol = "smtps"
+			case "587":
+				MailService.Protocol = "smtp+startls"
+			default:
+				log.Fatal("unable to infer unspecified mailer.PROTOCOL from mailer.SMTP_PORT = \"%s\"", MailService.SMTPPort)
+			}
+		}
+	}
+
+	// we want to warn if users use SMTP on a non-local IP;
+	// we might as well take the opportunity to check that it has an IP at all
+	ips := tryResolveAddr(MailService.SMTPAddr)
+	if MailService.Protocol == "smtp" {
+		for _, ip := range ips {
+			if !ip.IsLoopback() {
+				log.Warn("connect via insecure SMTP to non-local address")
```

Comment:
This error doesn't sound user-friendly.

Reply:
Not sure how exactly it should be reworded. Open to suggestions.

Reply:
Maybe: `Connecting over insecure SMTP protocol to a non-local address`

Reply:
Oh, I see what you mean now. Yeah, I can fix that

Reply:
Reworded to "connecting over insecure SMTP protocol to non-local address is not recommended"